File: Chess/Location.cs
using System.Collections.Generic;

namespace Chess
{
	public class Location
	{
		public readonly int X, Y;

		public Location(int x, int y)
		{
			X = x;
			Y = y;
		}

		public bool InBoard
		{
			get { return X >= 0 && X <= 7 && Y >= 0 && Y <= 7; }
		}

		public static IEnumerable<Location> AllBoard()
		{
			for (var y = 0; y < 8; y++)
				for (var x = 0; x < 8; x++)
					yield return new Location(x, y);
		}

		public override string ToString()
		{
			return new string((char) ('a' + X), 1) + Y;
		}

		public override bool Equals(object obj)
		{
			var other = obj as Location;
			if (other == null) return false;
			return other.X == X && other.Y == Y;
		}

		public override int GetHashCode()
		{
			unchecked
			{
				return (X*397) ^ Y;
			}
		}
	}
}

File: python/Chess/Solved/location.py
class Location:
    def __init__(self, x: int, y: int):
        self.x = x
        self.y = y

    def __str__(self):
        return f"({self.x}, {self.y})"

    def __eq__(self, other):
        if isinstance(other, Location):
            return self.x == other.x and self.y == other.y
        return False

    def __mul__(self, other):
        if isinstance(other, Location):
            return Location(self.x * other.x, self.y * other.y)
        elif isinstance(other, int):
            return Location(self.x * other, self.y * other)
        return self

    def __add__(self, other):
        if isinstance(other, Location):
            return Location(self.x + other.x, self.y + other.y)
        else:
            return self
